zrml/prediction-markets/src/lib.rs

```diff
@@ -1349,6 +1350,338 @@ mod pallet {
+
+            let get_new_period = |block_period,
+                                  time_frame_period|
+             -> Result<
+                MarketPeriod<T::BlockNumber, MomentOf<T>>,
+                DispatchError,
+            > {
+                match &market.period {
+                    MarketPeriod::Block(range) => {
+                        let close_time = now_block.saturating_add(block_period);
+                        ensure!(close_time < range.end, Error::<T>::EarlyCloseRequestTooLate);
+                        Ok(MarketPeriod::Block(range.start..close_time))
+                    }
+                    MarketPeriod::Timestamp(range) => {
+                        let close_time = now_time.saturating_add(time_frame_period);
+                        ensure!(close_time < range.end, Error::<T>::EarlyCloseRequestTooLate);
+                        Ok(MarketPeriod::Timestamp(range.start..close_time))
+                    }
+                }
+            };
+            let new_period = match &market.early_close {
+                None => {
+                    let (block_period, time_frame_period) = if is_authorized {
+                        // fat finger protection
+                        (
+                            T::CloseEarlyProtectionBlockPeriod::get(),
+                            T::CloseEarlyProtectionTimeFramePeriod::get(),
+                        )
+                    } else {
+                        let market_creator =
+                            market_creator.ok_or(Error::<T>::RequesterNotCreator)?;
+                        let close_request_bond = T::CloseEarlyRequestBond::get();
+
+                        T::AssetManager::reserve_named(
+                            &Self::reserve_id(),
+                            Asset::Ztg,
+                            &market_creator,
+                            close_request_bond,
+                        )?;
+
+                        <zrml_market_commons::Pallet<T>>::mutate_market(&market_id, |market| {
+                            market.bonds.close_request =
+                                Some(Bond::new(market_creator, close_request_bond));
+                            Ok(())
+                        })?;
+
+                        (T::CloseEarlyBlockPeriod::get(), T::CloseEarlyTimeFramePeriod::get())
+                    };
+
+                    get_new_period(block_period, time_frame_period)?
+                }
+                Some(p) => {
+                    ensure!(is_authorized, Error::<T>::OnlyAuthorizedCanScheduleEarlyClose);
+
+                    match p.state {
+                        // in these case the market period got already reset to the old period
+                        EarlyCloseState::Disputed => {
+                            if Self::is_close_dispute_bond_pending(&market_id, &market, false) {
+                                Self::repatriate_close_dispute_bond(&market_id, &market.creator)?;
```

Comment:
Why does the market creator get it? I think it makes more sense to let the `Slash` handler deal with it.

Reply:
Good question. Was just an idea. Indeed, we could fund the treasury with it. On the other hand, since the AC consists of members of Zeitgeist, we would have an incentive conflict here. Because Zeitgeist would receive the bond if the AC accepts the close request by the market creator. I know it's not that huge of a effect, but the next best option is to just give it to the market creator, because the market creator had to wait (opportunity cost) with a reserved bond.

Reply:
Right as of now the AC is kind of centralized, however that should not be an assumption. I don't know, it feels like the market creator is incentivized to mislead others into making false disputes.

Reply:
Another option would be to `withdraw` or burn the bond.

> market creator is incentivized to mislead others into making false disputes

If the market creator would ever mislead, he would get slashed. Can you give an example how this would look in practice? It seems to me that this is a scenario that would never happen.

Reply:
@maltekliemann

Reply:
I think ideally we should use the same rules as we do for disputes on oracle reports, but I don't have any strong opinions one way or the other.

As for the game theory, I see the point, except that I'm not really sure how you'd mislead people into making incorrect disputes. I think you could level the same criticism against the decentralized court.